Makefile

```diff
@@ -8,30 +8,7 @@ ifneq ($(wildcard ./.git/),)
 VERSION := $(shell ${GIT} describe --tags)
 endif
 
-ifeq (,${SYSTEMD})
-# Check for systemctl to avoid discrepancies on systems, where
-# systemd is installed, but systemd.pc is in another package
-systemctl := $(shell command -v ${SYSTEMCTL} >/dev/null && echo systemctl)
-ifeq (systemctl,${systemctl})
-SYSTEMD := 1
-else
-SYSTEMD := 0
-endif
-endif
-
-SERVICEDIR_DBUS ?= $(shell $(PKG_CONFIG) dbus-1 --variable=session_bus_services_dir)
-SERVICEDIR_DBUS := ${SERVICEDIR_DBUS}
-ifeq (,${SERVICEDIR_DBUS})
-$(error "Failed to query $(PKG_CONFIG) for package 'dbus-1'!")
-endif
-
-ifneq (0,${SYSTEMD})
-SERVICEDIR_SYSTEMD ?= $(shell $(PKG_CONFIG) systemd --variable=systemduserunitdir)
-SERVICEDIR_SYSTEMD := ${SERVICEDIR_SYSTEMD}
-ifeq (,${SERVICEDIR_SYSTEMD})
-$(error "Failed to query $(PKG_CONFIG) for package 'systemd'!")
-endif
-endif
+SYSTEMD ?= $(shell $(SYSTEMDAEMON) --version >/dev/null || echo 0)
```

Comment:
On my system (Arch) the binary `systemd` is not in my path. I would have to run `/usr/lib/systemd/systemd` for it to work. I'm not sure what a portable way to check it is. You would probably have to check at least `/usr/lib/systemd/systemd and `systemd`.
See also: https://bugs.archlinux.org/task/35195

Reply:
Thanks for pointing that out. I just checked and found that '/usr/bin/systemd' is symlinked to '/lib/systemd/systemd' on Ubuntu, so the real executable is not in PATH either. I think the change I just added resolves this.